src/components/movie-detail.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from "react";
+import '../movie.css';
+import * as api from '../api';
+import { RouteComponentProps } from "react-router";
+
+type Genres = {
+    id: number;
+    name: string;
+}
+type Movie = {
+    id: string;
+    title: string;
+    release_date: string;
+    poster_path: string;
+    vote_average: number;
+    overview: string;
+    genres: Genres[];
+}
+
+type State = {
+    selected_movie?: Movie;
+}
+type Props = RouteComponentProps<{movieId: string}>;
+
+class MovieDetail extends Component<Props> {
+
+    state: State = { 
+    }
+
+    componentDidMount() {
+        api.getMovieDetail(this.props.match.params.movieId).then((movie) => {
+          this.setState({selected_movie: movie})
+        })
+    }
+
+    render() {
+        let releaseDate = new Date((this.state.selected_movie) ? this.state.selected_movie.release_date:"").getFullYear();
+        return (
+            <div className="movie-detail-container">
+                {
+                    (this.state.selected_movie)
+                        ?   <div className="movie-detail">
+                                <h1>{this.state.selected_movie.title}</h1>
+                                <div className="inlining">
+                                    <div className="rating">{this.state.selected_movie.vote_average}/<span>10</span></div>
+                                    <div className="genres">{this.state.selected_movie.genres.map((genre) => <div key={genre.id} className="genre">{genre.name}</div>)}</div>
+                                    <div className="release-date">{releaseDate}</div>
```

Comment:
Do `getFullYear` logic here. Because you dont need to check if `selected_movie ` exists here anymore